docker/test/integration/test_http.py

```diff
@@ -0,0 +1,36 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the \"License\"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an \"AS IS\" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from minifi import *
+from minifi.test import *
+
+
+def test_invoke_listen():
+    """
+    Verify sending using InvokeHTTP to a receiver using ListenHTTP.
+    """
+
+    invoke_flow = (GetFile('/tmp/input') >> LogAttribute() >>
+                   InvokeHTTP('http://minifi-listen:8080/contentListener', method='POST'))
+
+    listen_flow = ListenHTTP(8080) >> LogAttribute() >> PutFile('/tmp/output')
+
+    with DockerTestCluster(SingleFileOutputValidator('test')) as cluster:
+
+        cluster.put_test_data('test')
+        cluster.deploy_flow(listen_flow, name='minifi-listen')
+        cluster.deploy_flow(invoke_flow, name='minifi-invoke')
+
+        assert cluster.check_output()
```

Comment:
This seems to be failing for me in OS X with Docker Mac.  Output below:

```
DEBUG:urllib3.connectionpool:http://localhost:None "DELETE /v1.30/networks/5f6cae20d1721e4ba01039fddabedc627532b5ec698897598cf6f18d2eed21d4 HTTP/1.1" 204 0
FAILED

==================================================================================== FAILURES ====================================================================================
_______________________________________________________________________________ test_invoke_listen _______________________________________________________________________________

    def test_invoke_listen():
        """
        Verify sending using InvokeHTTP to a receiver using ListenHTTP.
        """

        invoke_flow = (GetFile('/tmp/input') >> LogAttribute() >>
                       InvokeHTTP('http://minifi-listen:8080/contentListener', method='POST'))

        listen_flow = ListenHTTP(8080) >> LogAttribute() >> PutFile('/tmp/output')

        with DockerTestCluster(SingleFileOutputValidator('test')) as cluster:

            cluster.put_test_data('test')
            cluster.deploy_flow(listen_flow, name='minifi-listen')
            cluster.deploy_flow(invoke_flow, name='minifi-invoke')

>           assert cluster.check_output()
E           assert False
E            +  where False = <bound method DockerTestCluster.check_output of <minifi.test.DockerTestCluster object at 0x102a51e50>>()
E            +    where <bound method DockerTestCluster.check_output of <minifi.test.DockerTestCluster object at 0x102a51e50>> = <minifi.test.DockerTestCluster object at 0x102a51e50>.check_output

../docker/test/integration/test_http.py:36: AssertionError
====================================================================== 1 failed, 2 passed in 22.31 seconds =======================================================================
make[3]: *** [CMakeFiles/docker-verify] Error 1
make[2]: *** [CMakeFiles/docker-verify.dir/all] Error 2
make[1]: *** [CMakeFiles/docker-verify.dir/rule] Error 2
make: *** [docker-verify] Error 2
```

Reply:
The http tests are dependent on the phrocker/MINIFI-339 branch to pass, and possibly some of the other PRs I put in. There is unfortunately a bit of inter-dependence going on right now between the various branches.